### PR TITLE
GH actions: modify workflows builder and build-packages to build also arm64 packages

### PR DESCRIPTION
**Pull request by romeroalx**
### Short description
Update `builder.yml` and `build-packages.yml` to build `arm64` packages. Runners from Ubicloud are used for the `arm64` builds.

Removed symlinks for making a distinction between `amd64` and `arm64` Dockerfiles, letting docker decide the architecture of an image based on the host where the build is run.

In addition:
- Source hashes have been updated removing the list of packages from its generation
- Updated the path where attestations are stored to use the "normalized" product name.
- Usage of Ubuntu-22.04 runners for `builder.yml` and `build-packages.yml`

Test run for `builder.yml`: [https://github.com/romeroalx/pdns/actions/runs/9209542101](https://github.com/romeroalx/pdns/actions/runs/9209542101)

Test run for building packages manually for a product (`builder-dispatch.yml`): [https://github.com/romeroalx/pdns/actions/runs/9224283242](https://github.com/romeroalx/pdns/actions/runs/9224283242)

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Review by rgacogne — APPROVED** [review]
Looks good to me!

> Source hashes have been updated removing the list of packages from its generation

Makes sense to me, the tarball is actually not generated in the layer for which we generate the packages list anyway, which is something we might need to fix in the future.

> Updated the path where attestations are stored to use the "normalized" product name.

:+1:

File: builder-support/dockerfiles/Dockerfile.target.centos-8 [hunk 1]
```diff
@@ -9,12 +9,6 @@ FROM centos:8 as dist-base
 @IF [ ${BUILDER_TARGET} = centos-8-stream ]
 FROM quay.io/centos/centos:stream8 as dist-base
 @ENDIF
-@IF [ ${BUILDER_TARGET} = centos-8-amd64 ]
-FROM amd64/centos:8 as dist-base
```
Comment (Habbie) [inline builder-support/dockerfiles/Dockerfile.target.centos-8:13]: this reminds me that `centos:8` is a dead target on every arch, and the only thing still alive in -this- Dockerfile is 8-stream. But we can clean that up separately.

**Review by Habbie — APPROVED** [review]
approved with an unrelated note

File: builder-support/dockerfiles/Dockerfile.target.centos-8 (hunk 1 above)
Comment (romeroalx) [reply builder-support/dockerfiles/Dockerfile.target.centos-8:13]: Ok! I will do it now

Comment (romeroalx) [reply builder-support/dockerfiles/Dockerfile.target.centos-8:13]: Done. I also changed the endpoint from where packages are retrieved for `centos:8-stream` from `mirrorlist` to `vault`.

BTW, **Centos 8 stream** reached end-of-builds a few days ago. Maybe it is just better to change it to `el-8`.

**Comment by romeroalx** [comment]
Rebased

File: builder-support/dockerfiles/Dockerfile.target.centos-8 (hunk 1 above)
Comment (Habbie) [reply builder-support/dockerfiles/Dockerfile.target.centos-8:13]: > BTW, **Centos 8 stream** reached end-of-builds a few days ago.

oh darn, I missed that.

> Maybe it is just better to change it to `el-8`.

yes, indeed, all traces of centos 8 (normal and stream) can go then.

**Pull request merged**